File: src/ch03/code_2.py
from typing import List

from rich import print


def linear_search(nums: List[int], target: int) -> int:
    for idx, num in enumerate(nums):
        if num == target:
            return idx
    return -1


if __name__ == '__main__':
    nums: List[int] = [3, 4, 1, 2, 5, 7]
    target: int = 5
    result: int = linear_search(nums=nums, target=target)
    print(result)

File: src/ch03/code_2.rs
fn linear_search(nums: Vec<i32>, target: i32) -> i32 {
    for (idx, num) in nums.iter().enumerate() {
        if *num == target {
            return idx as i32;
        }
    }
    -1
}


fn main() {
    let nums: Vec<i32> = vec![3, 4, 1, 2, 5, 7];
    let target: i32 = 5;
    let result: i32 = linear_search(nums, target);

    println!("{}", result);
}